src/java-interop/java-interop-gc-bridge-mono.c

```diff
@@ -155,11 +160,45 @@ ji_realpath (const char *path)
 	if (path == NULL)
 		return NULL;
 
+#ifndef WINDOWS
 	char   *rp = realpath (path, NULL);
 	if (rp == NULL) {
 		return strdup (path);
 	}
 	return rp;
+#else
+#define BUFSIZE 4096
+
+	wchar_t *wpath = utf8_to_utf16 (path);
+	char *fullpath = NULL;
+	wchar_t buffer [BUFSIZE];
+	DWORD retval = GetFullPathNameW (wpath, BUFSIZE, buffer, NULL);
+
+	do {
+		if (retval == 0)
+			break;
+
+		if (retval >= BUFSIZE) {
```

Comment:
Leaving behind the advisability of having an 8KB buffer on the stack...

This logic should be inverted so that we can drop a level of indentation:

```c
if (retval < BUFSIZE) {
    fullpath = utf16_to_utf8 (buffer);
    break;
}
wchar_t *allocated_buffer = xmalloc (sizeof(wchar_t)*retval);
...
```

Reply:
Good point. Updated.

I have also lowered the stack buffer size to 512*2 which is twice the MAX_PATH and should cover most cases.